src/gyazo.rb

```diff
@@ -1,130 +1,54 @@
 #!/usr/bin/env ruby
 
-require 'net/http'
 require 'open3'
-require 'openssl'
 require 'json'
 require 'yaml'
+require_relative 'gyazo_client'
 
 # setting
-configfile = "#{ENV['HOME']}/.gyazo.config.yml"
-config = {}
-if File.exist?(configfile) then
-  config = YAML.load_file(configfile)
-end
-
-browser_cmd = config['browser_cmd'] || 'xdg-open'
-clipboard_cmd = config['clipboard_cmd'] || 'xclip'
-clipboard_opt = config['clipboard_opt'] || '-sel clip'
-host = config['host'] || 'upload.gyazo.com'
-cgi = config['cgi'] || '/upload.cgi'
-ua = config['ua'] || 'Gyazo/1.2'
-http_port = config['http_port'] || 443
-use_ssl = config['use_ssl'] == nil ? 'true' : config['use_ssl']
-
-# get id
-idfile = ENV['HOME'] + "/.gyazo.id"
-
-id = ''
-if File.exist?(idfile) then
-  id = File.read(idfile).chomp
-end
+config_file = "#{ENV['HOME']}/.gyazo.config.yml"
+opts = {}
+opts = YAML.load_file(config_file) if File.exist?(config_file)
 
-# get active window name
+gyazo_client = GyazoClient.new(opts)
```

Comment:
[IMO] I think it's better that it passes yaml path to GyazoClient then YAML load in `GyazoClient`'s code